modules/learn/pages/data/transactions.adoc

```diff
@@ -0,0 +1,56 @@
+= Distributed ACID Transactions
+
+[abstract]
+_Distributed ACID Transactions_ guarantee that multiple documents can be updated atomically.
+
+[#transactions-and-couchbase-server]
+== Transactions and Couchbase Server
+
+_Transactions_ are operations that ensure that when multiple discrete data-elements need to be modified such that only the successful modification of all justifies the modification of any, _either_ all the modifications do occur successfully; _or_ none of them occurs.
+Transactions are implemented by Couchbase Server as _Distributed ACID Transactions_.
+
+A Distributed ACID Transaction can be performed on any number of documents.
+For each document, the transaction makes a copy of the document-data, and saves the copy temporarily in the xref:learn:data/extended-attributes.adoc[Extended Attributes] area of the document's meta data.
+Updates and modifications are then made to the key-value pairs contained by this meta data copy: consequently, the changes that occur during the transaction prior to commitment &#8212; these constituting _uncommitted_ (or _dirty_) data &#8212; are non-readable by any operation other than this transaction itself.
```

Comment:
This makes it sound like there's separate operations to copy the data into xattrs, _then_ apply modifications.  That sounds slow.  Instead, the app gets a document inside a transaction.  They make all modifications to that in-memory of the version of the document.  Then they call ctx.replace(), which puts that modified copy into the xattrs.

Reply:
How about: 'For each document, the transaction makes changes in memory; then puts the modified copy of the document into the Extended Attributes area of the document's meta data. Consequently, the changes that occur during the transaction prior to commitment - these constituting 'uncommitted' or 'dirty' data - are non-readable by any operation other than this transaction itself.'